bundle/permissions/permission_diagnostics_test.go

```diff
@@ -22,14 +22,31 @@ func TestPermissionDiagnosticsApplySuccess(t *testing.T) {
 	require.NoError(t, diags.Error())
 }
 
+func TestPermissionDiagnosticsEmpty(t *testing.T) {
+	b := mockBundle(nil)
+
+	diags := permissions.PermissionDiagnostics().Apply(context.Background(), b)
```

Comment:
I recommend using `bundle.Apply` because that also includes the conversion to and from the dynamic representation. It more accurately emulates how this mutator will be applied and is what we use in unit tests in other packages.